Core/clim-basic/extended-streams/stream-input.lisp

```diff
@@ -43,6 +44,104 @@
+    with timeout-time = (and timeout (+ timeout (now)))
+    when (stream-gesture-available-p stream)
+      do (return-from stream-input-wait t)
+    do (multiple-value-bind (available reason)
+           (event-listen-or-wait stream :timeout timeout
+                                        :wait-function wait-fun)
+         (when (and (null available) (eq reason :timeout))
+           (return-from stream-input-wait (values nil :timeout)))
+         (when-let ((event (event-read-no-hang stream)))
+           (handle-event (event-sheet event) event))
+         (when timeout
+           (setf timeout (compute-decay timeout-time nil)))
+         (when (funcall input-wait-test stream)
+           (return-from stream-input-wait
+             (values nil :input-wait-test))))))
+
+(defmethod stream-listen ((stream input-stream-kernel))
+  (stream-input-wait stream))
+
+(defmethod stream-process-gesture ((stream input-stream-kernel) gesture type)
+  (declare (ignore stream))
+  (values gesture type))
+
+(defmethod stream-read-gesture ((stream input-stream-kernel)
+                                &key timeout peek-p
+                                  (input-wait-test *input-wait-test*)
+                                  (input-wait-handler *input-wait-handler*)
+                                  (pointer-button-press-handler
+                                   *pointer-button-press-handler*))
+  (when peek-p
+    (return-from stream-read-gesture
+      (stream-gesture-available-p stream)))
+  (let ((*input-wait-test* input-wait-test)
+        (*input-wait-handler* input-wait-handler)
+        (*pointer-button-press-handler* pointer-button-press-handler)
+        (input-buffer (stream-input-buffer stream)))
+    (flet ((process-gesture (raw-gesture)
+             (when (and pointer-button-press-handler
+                        (typep raw-gesture 'pointer-button-press-event))
+               (funcall pointer-button-press-handler stream raw-gesture))
+             (when-let ((gesture (stream-process-gesture stream raw-gesture t)))
+               (return-from stream-read-gesture gesture))))
+      (loop
+        (multiple-value-bind (available reason)
+            (stream-input-wait stream :timeout timeout
+                                      :input-wait-test input-wait-test)
+          (if available
+              (process-gesture (event-queue-read input-buffer))
+              (case reason
+                (:input-wait-test
+                 (maybe-funcall input-wait-handler stream))
+                (:timeout
+                 (return-from stream-read-gesture (values nil :timeout)))
+                #+ (or)
+                ;; EOF may be returned by input streams backed by a string
+                ;; (see the "XXX Evil hack" below), but that should not happen
+                ;; for input-stream-kernel.
+                (:eof
+                 (return-from stream-read-gesture (values nil :eof)))
```

Comment:
this clause is commented out, so the error *will* be signaled. I'd rather keep this as is.

Reply:
Yes, I was suggesting that the clause is not needed (even wrong according to the comment).

Reply:
as I've said, I'd rather keep it commented because there are streams, which return EOF (most notably the evil hack stream), so this keyword happens in a similar context, so having an explicitly commented out clause signifies the fact, that this condition should not happen here despite being possible in other methods of this function.

Reply:
I think an extended comment without the commented out code would explain this more clearly.

Reply:
commented out block of code is a kind of an extended comment. I've put the comment before the case however I find the previous version better.

Reply:
I'm not convinced but that shouldn't block the merge any longer.